django/db/models/expressions.py

```diff
@@ -158,6 +158,8 @@ class BaseExpression:
     filterable = True
     # Can the expression can be used as a source expression in Window?
     window_compatible = False
+    # Can the expression be used as a database default value?
+    allowed_default = False
```

Comment:
Do we need a new flag? IMO it's unnecessary. It should be feasible to detect this automatically in the system check, e.g. with `_get_expr_references(expression)`.

Reply:
We've discussed this at DjangoCon Europe 2022 and we think this is a good approach.